go-controller/pkg/ovnwebhook/nodeadmission.go

```diff
@@ -0,0 +1,169 @@
+package ovnwebhook
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	hotypes "github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"golang.org/x/exp/maps"
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	nodeutil "k8s.io/component-helpers/node/util"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type checkNodeAnnot func(v annotationChange, nodeName string) error
+
+// commonNodeAnnotationChecks holds annotations allowed for ovnkube-node:<nodeName> users in non-IC and IC environments
```

Comment:
comment on what the value is, it looks like a function that further restricts modifications for this allowed type?